boards: arm: efr32_radio: Add I2C0 node

Arunmanialagarsamy opened the pull request:
Add i2c0 pincntrl for efr32_radio_brd4161a

ArunmaniAlagarsamy2710 committed:
boards: silabs: Add I2C node

Add i2c pincntrl to silabs boards that supports pinctrl api.

Signed-off-by: Arunmani Alagarsamy <arunmani.a@capgemini.com>

nordicjm reviewed (dismissed):
Needs porting to hwmv2

fkokosinski commented on boards/arm/efr32_radio/efr32_radio_brd4161a.dts:
```diff
@@ -68,3 +68,9 @@
 };
+
+&i2c0 {
+	location-sda = <GECKO_LOCATION(16) GECKO_PORT_C GECKO_PIN(11)>;
```
Using `location-*` properties is deprecated. You should convert these changes so that they use the Pinctrl API: https://docs.zephyrproject.org/latest/hardware/pinctrl/index.html

Arunmanialagarsamy replied:
https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/i2c/i2c_gecko.c
They didnt use the pinctrl api.
Do you want me to implement pinctrl apis in i2c_gecko.c driver?

fkokosinski replied:
That would be required for the Gecko I2C driver to use pinmuxing. You could base your work off of the Gecko UART driver: https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/serial/uart_gecko.c

gmarull commented on drivers/i2c/i2c_gecko.c:
```diff
@@ -24,11 +25,16 @@ LOG_MODULE_REGISTER(i2c_gecko);
 #define DEV_BASE(dev) ((I2C_TypeDef *)((const struct i2c_gecko_config *const)(dev)->config)->base)
 
 struct i2c_gecko_config {
+#ifdef CONFIG_PINCTRL
+	const struct pinctrl_dev_config *pcfg;
+#endif
 	I2C_TypeDef *base;
 	CMU_Clock_TypeDef clock;
 	uint32_t bitrate;
+#ifndef CONFIG_PINCTRL
 	struct soc_gpio_pin pin_sda;
 	struct soc_gpio_pin pin_scl;
+#endif
```
please, make pinctrl mandatory, old methods should actually be removed as deprecation period is gone.

fkokosinski reviewed (changes requested):
These changes seem to introduce a regression for me. Tested with:

```
west build -p -b efr32xg24_dk2601b samples/subsys/shell/shell_module -- -DCONFIG_I2C=y -DCONFIG_I2C_SHELL=y
```

With your changes, I get no UART output.

fkokosinski commented:
I've unassigned myself since I'm no longer maintaining SiLabs platorms (see https://github.com/zephyrproject-rtos/zephyr/pull/71219).

gmarull commented on drivers/i2c/i2c_gecko.c:
```diff
@@ -182,7 +151,11 @@ static int i2c_gecko_init(const struct device *dev)
+	error = pinctrl_apply_state(config->pcfg, PINCTRL_STATE_DEFAULT);
+	if (error < 0) {
+		LOG_ERR("Failed to configure I2C pins err[%d]", error);
+		return -EINVAL;
```
please, propagate error codes

gmarull reviewed (changes requested):
split changes better, eg i2c driver changes into its own commit

ArunmaniAlagarsamy2710 committed:
drivers: pinctrl: pinctrl_gecko: Add support for using pinctrl api

This update integrates I2C support with the pinctrl_configure_pins api
within the pinctrl_gecko driver.

Signed-off-by: Arunmani Alagarsamy <arunmani.a@capgemini.com>

ArunmaniAlagarsamy2710 committed:
drivers: i2c: i2c_gecko: Refactor driver to use  pinctrl api

Deprecate the use of location_* properties in the i2c_gecko driver
and migrate to the pinctrl api for enhanced maintainability and
compliance with current standards.

Signed-off-by: Arunmani Alagarsamy <arunmani.a@capgemini.com>

Arunmanialagarsamy commented:
@gmarull Gentle ping!

Arunmanialagarsamy commented:
@tgorochowik @fkokosinski waiting for your approval, please review it
